src/EventGrid/EventGrid/help/Get-AzEventGridDomain.md

```diff
@@ -14,63 +14,89 @@ Gets the details of an Event Grid domain, or gets a list of all Event Grid domai
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
 ```
 
 ### DomainNameParameterSet
 ```
-Get-AzEventGridDomain [-ResourceGroupName] <String> [-Name] <String> [-DefaultProfile <IAzureContextContainer>]
- [<CommonParameters>]
+Get-AzEventGridDomain [-ResourceGroupName] <String> [-Name] <String> [-ODataQuery <String>] [-Top <Int32>]
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
 ```
 
 ### ResourceIdEventSubscriptionParameterSet
 ```
-Get-AzEventGridDomain [-ResourceId] <String> [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+Get-AzEventGridDomain [-ResourceId] <String> [-ODataQuery <String>] [-Top <Int32>]
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+### NextLinkParameterSet
+```
+Get-AzEventGridDomain [-NextLink <String>] [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
 ```
 
 ## DESCRIPTION
 The Get-AzEventGridDomain cmdlet gets either the details of a specified Event Grid domain, or a list of all Event Grid domains in the current Azure subscription.
 If the domain name is provided, the details of a single Event Grid domain is returned.
-If the domain name is not provided, a list of domains is returned.
+If the domain name is not provided, a list of domains is returned. The number of elements returned in this list is controlled by the Top parameter. If the Top value is not specified or $null, the list will contain all the domains items. Otherwise, Top will indicate the maximum number of elements to be returned in the list.
+If more domains are still available, the value in NextLink should be used in the next call to get the next page of domains.
+Finally, ODataQuery parameter is used to perform filtering for the search results. The filtering query follows OData syntax using the Name property only. The supported operations include: CONTAINS, eq (for equal), ne (for not equal), AND, OR and NOT.
 
 ## EXAMPLES
 
 ### Example 1
-```
+```powershell
 PS C:\> Get-AzEventGridDomain -ResourceGroup MyResourceGroupName -Name Domain1
 ```
 
 Gets the details of Event Grid domain \`Domain1\` in resource group \`MyResourceGroupName\`.
 
 ### Example 2
-```
+```powershell
 PS C:\> Get-AzEventGridDomain -ResourceId "/subscriptions/$subscriptionId/resourceGroups/MyResourceGroupName/providers/Microsoft.EventGrid/domains/Domain1"
 ```
 
 Gets the details of Event Grid domain \`Domain1\` in resource group \`MyResourceGroupName\`.
 
 ### Example 3
-```
+```powershell
 PS C:\> Get-AzEventGridDomain -ResourceGroup MyResourceGroupName
 ```
 
-List all the Event Grid domains in resource group \`MyResourceGroupName\`.
+List all the Event Grid domains in resource group \`MyResourceGroupName\` without pagination.
 
 ### Example 4
+```powershell
+$odataFilter = "Name ne 'ABCD'"
+PS C:\> $result = Get-AzEventGridDomain -ResourceGroup MyResourceGroupName -Top 10 -ODataQuery $odataFilter
+PS C:\> Get-AzEventGridDomain $result.NextLink
```

Comment:
@ahamad-MS for the examples you're adding, would you mind also including the corresponding output that the user would see if they ran the corresponding command?

Reply:
Is there specific pattern or format for adding the output here. I checked multiple other cmdlets (for storage) and I don't seem to see such a pattern applied. I would like to address this comment for all the examples in the help to have consistent behavior but this will require separate changelist with additional info. Please feel free to open an issue and assign it to me and I will address it in separate PR if you don't mind.

Reply:
@ahamad-MS in general, output from cmdlets should follow the below pattern:

- If there is a single object returned from a cmdlet, it should be in a list view
- If there is more than one object returned from a cmdlet, it should be in a table view

Information about how to properly format these outputs can be found [here](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/generate-format.ps1xml-file.md).

Examples should contain the corresponding output that users would see if they were to run the command in the example -- this allows users to see properties (with sample values) are returned with the object. You can check out the [KeyVault help folder](https://github.com/Azure/azure-powershell/tree/master/src/KeyVault/KeyVault/help) for examples of existing cmdlets with outputs displayed in their examples.

Reply:
Thx for the info. Per our offline discussion, we will address this comment in separate checkin.